homeassistant/components/smarty/sensor.py

```diff
@@ -0,0 +1,237 @@
+
+    def update(self) -> None:
+        """Update state."""
+        _LOGGER.debug('Updating sensor %s', self._name)
+        self._state = self._smarty.outdoor_air_temperature
+
+
+class SupplyFanSpeedSensor(SmartySensor):
+    """Supply Fan Speed RPM."""
+
+    def __init__(self, name, smarty):
+        """Supply Fan Speed RPM Init."""
+        super().__init__(name=name + ' Supply Fan Speed',
+                         device_class=DEVICE_CLASS_RPM,
+                         unit_of_measurement=RPM,
+                         smarty=smarty)
+
+    def update(self) -> None:
+        """Update state."""
+        _LOGGER.debug('Updating sensor %s', self._name)
+        self._state = self._smarty.supply_fan_speed
+
+
+class ExtractFanSpeedSensor(SmartySensor):
+    """Extract Fan Speed RPM."""
+
+    def __init__(self, name, smarty):
+        """Extract Fan Speed RPM Init."""
+        super().__init__(name=name + ' Extract Fan Speed',
+                         device_class=DEVICE_CLASS_RPM,
+                         unit_of_measurement=RPM,
+                         smarty=smarty)
+
+    def update(self) -> None:
+        """Update state."""
+        _LOGGER.debug('Updating sensor %s', self._name)
+        self._state = self._smarty.extract_fan_speed
+
+
+class SystemStateSensor(SmartySensor):
+    """System State Sensor."""
+
+    def __init__(self, name, smarty):
+        """Alarm Sensor Init."""
+        super().__init__(name=name + ' System State',
+                         device_class=DEVICE_CLASS_STATUS,
+                         smarty=smarty)
+
+    def update(self) -> None:
+        """Update state."""
+        _LOGGER.debug('Updating sensor %s', self._name)
+        self._state = self._smarty.system_state
+
+
+class AlarmSensor(SmartySensor):
+    """Alarm Sensor."""
+
+    def __init__(self, name, smarty):
+        """Alarm Sensor Init."""
+        super().__init__(name=name + ' Alarm',
+                         device_class=DEVICE_CLASS_ALARM,
+                         smarty=smarty)
+
+    def update(self) -> None:
+        """Update state."""
+        _LOGGER.debug('Updating sensor %s', self._name)
+        self._state = STATE_ON if self._smarty.alarm else STATE_OFF
+
+
+class WarningSensor(SmartySensor):
+    """Warning Sensor."""
+
+    def __init__(self, name, smarty):
+        """Warning Sensor Init."""
+        super().__init__(name=name + ' Warning',
+                         device_class=DEVICE_CLASS_ALARM,
+                         smarty=smarty)
+
+    def update(self) -> None:
+        """Update state."""
+        _LOGGER.debug('Updating sensor %s', self._name)
+        self._state = STATE_ON if self._smarty.warning else STATE_OFF
+
+
+class FilterDaysLeftSensor(SmartySensor):
+    """Filter Days Left."""
+
+    def __init__(self, name, smarty):
+        """Filter Days Left Init."""
+        super().__init__(name=name + ' Filter Days Left',
+                         device_class=DEVICE_CLASS_DAYS,
```

Comment:
Consider changing this to use the timestamp device class, and return an absolute time point instead for state. That's a better user experience.

Reply:
Not sure I understand when you mean. The device is returning a number between 0-90 which is in DAYS remaining.

Reply:
Yes, so convert that to an absolute time stamp by comparing to time now here. Using the timestamp device class allows the user to choose a relative time representation in the GUI, and it puts less strain on our state machine since an absolute time stamp will generate fewer state updates.